platforms/enterprise/enterprise-operations/src/main/java/org/gradle/internal/configurationcache/ConfigurationCacheCheckFingerprintBuildOperationType.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.configurationcache;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+import java.util.List;
+
+/**
+ * Details about the configuration cache fingerprint check operation. This operation is only emitted when the configuration cache entry is available.
+ *
+ * @since 8.10
+ */
+public class ConfigurationCacheCheckFingerprintBuildOperationType implements BuildOperationType<ConfigurationCacheCheckFingerprintBuildOperationType.Details, ConfigurationCacheCheckFingerprintBuildOperationType.Result> {
+    /**
+     * Build operation details.
+     *
+     * @since 8.10
+     */
+    public interface Details {}
+
+    /**
+     * Result of the fingerprint check. The fingerprint can be:
+     * <ul>
+     *     <li><b>Valid</b>: the cached configuration can be fully reused.
+     *     Both {@link #getBuildInvalidationReasons()} and {@link #getProjectInvalidationReasons()} are empty.</li>
+     *
+     *     <li><b>Partially invalidated</b>: some projects are invalidated and have to be reconfigured.
+     *     The rest of the configuration data is still reused.
+     *     The {@link #getBuildInvalidationReasons()} list is empty.
+     *     The {@link #getProjectInvalidationReasons()} list contains information about invalidated projects.</li>
+     *
+     *     <li><b>Fully invalidated</b>: some of the build-level inputs has changed. The entry is discarded and configuration phase reruns from scratch.
+     *     The {@link #getBuildInvalidationReasons()} list contains the invalidation reasons.
+     *     The {@link #getProjectInvalidationReasons()} list is empty.</li>
+     * </ul>
+     *
+     * @since 8.10
+     */
+    public interface Result {
+        /**
+         * Returns the list of the build fingerprint invalidation reasons. Can be empty if all build-level inputs are up-to-date.
+         * The entry may still be partially invalidated if any of the project-scoped fingerprints are invalid.
+         * <p>
+         * Not all invalidation reasons may be returned. However, if the entry is invalidated, at least one reason is listed.
+         * <p>
+         * The first invalidation in the list is what Gradle shows to the user.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<FingerprintInvalidationReason> getBuildInvalidationReasons();
```

Comment:
🤔 Would it make sense to have a simple list of `String`s here?

Reply:
+1

I'd love us to pass more structured messages for build scans, and get similar display capabilities like CC report provides. Though, we don't have to support it right away.

Reply:
Yeah, I think the problems API should support the kind of structured data, especially with relation to the locations it has anyway. Though I wouldn't do this here.